src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -495,10 +495,14 @@ void ReplicatedMergeTreeSinkImpl<async_insert>::consume(Chunk chunk)
             std::move(part_counters) /// profile_events_scope must be reset here.
         ));
     }
-
-    finishDelayedChunk(zookeeper);
     delayed_chunk = std::make_unique<ReplicatedMergeTreeSinkImpl::DelayedChunk>();
     delayed_chunk->partitions = std::move(partitions);
+    finishDelayedChunk(zookeeper);
+
+    if (streams > 0 && streams <= max_insert_delayed_streams_for_parallel_write)
```

Comment:
I think the root cause is we didn't set `last_block_is_duplicate` properly.
How about replacing line 512 with `last_block_is_duplicate=false`. If so, we don't need line 502-505, right?